sway-ir/src/verify.rs

```diff
@@ -468,6 +469,26 @@ impl<'a> InstructionVerifier<'a> {
         }
     }
 
+    fn verify_int_to_ptr(&self, value: &Value, ty: &Type) -> Result<(), IrError> {
+        // We want the source vaule to be an integer and the destination type to be a reference
+        // type.
+        let val_ty = value
+            .get_type(self.context)
+            .ok_or(IrError::VerifyIntToPtrUnknownSourceType)?;
+        if !matches!(val_ty, Type::Uint(64)) {
+            return Err(IrError::VerifyIntToPtrFromNonIntegerType(
+                val_ty.as_string(self.context),
+            ));
+        }
+        if ty.is_copy_type() {
```

Comment:
Interesting.  We never want to convert a `u64` to `ptr u64`?  I guess not yet, until we really have proper reference support.

Reply:
Yeah I didn't want think too hard about this just yet 😄